internal/cli/global_opts_test.go

```diff
@@ -25,40 +25,37 @@ import (
 func TestGlobalOpts_PreRunE(t *testing.T) {
 	t.Run("empty project ID", func(t *testing.T) {
 		o := &GlobalOpts{}
-		if err := o.PreRunE(); err != errMissingProjectID {
+		if err := o.PreRunE(o.ValidateProjectID); err != errMissingProjectID {
```

Comment:
you should probably change this test to be, `TestGlobalOpts_ValidateProjectID` and skip the `o.PreRunE` part given this is mostly testing the `ValidateProjectID` method

for `o.PreRunE` I'm not sure if we can have a proper unit tests here given what it does but feel free to try something

Reply:
like I said you don't need to call `o.PreRunE`
this can be just `if err := o.ValidateProjectID(); err != errMissingProjectID`, `PreRunE`, has no impact on this test so let's not mix them

Reply:
I am not sure if I agree here. In this way we are also testing `o.PreRunE`. in addition, `ValidateProjectID` and `ValidateOrgID` are used only with `o.PreRunE`. I understand that the generally idea is to not mix things together but I would argue that those two methods are bundled together.

Reply:
ValidateProjectID and ValidateOrgID, is a method of its own, so it should have its own set of unit tests, the same goes for `PreRunE`, it should be its own set of unit tests that should not depend on `ValidateProjectID` or `ValidateOrgID`. This is basic unit testing, you don't want your test to have external dependencies because if the external dependency changes means unrelated tests now need to be updated

The fact that `PreRunE` always calls ValidateProjectID or ValidateOrgID doesn't mean this will always be the case, we have now open the possibility to use `PreRunE` in a couple of places where we were not doing it before because it will always validate

So ideally a test for `PreRunE` would inject its own set of dependencies.